src/tests/suite/io_conf/daos_generate_io_conf.c

```diff
@@ -255,69 +258,113 @@ update_single(int index, daos_epoch_t eph, struct extent *extents,
-	int i;
+	char            rec_buf[512] = {0};
+	int             fetch_index;
+	int             rec_length = 0;
+	int             i;
 
 	D_ASSERT(index > 0);
 
 	fetch_index = rand() % index;
-	record = &records[fetch_index];
+	record      = &records[fetch_index];
 	for (i = 0; i < record->records_num; i++) {
 		struct array *array = &record->records[i].array;
-		int length;
-		char rec[64];
+		int           length;
+		char          rec[64];
 
-		length = sprintf(rec, "["DF_U64", "DF_U64"]%d",
-				 array->extent.start,
-				 array->extent.end, array->value);
+		length = sprintf(rec, "[" DF_U64 ", " DF_U64 "]%d",
+				 array->extent.start, array->extent.end,
+				 array->value);
 
 		sprintf(rec_buf + rec_length, "%s ", rec);
 		rec_length += length + 1;
 	}
 
-	if (rec_length == 0) {
-		sprintf(output_buf, "fetch --epoch "DF_U64" -s -v --value 0\n",
-			record->eph);
-	} else {
-		if (record->records[0].type == SINGLE)
-			sprintf(output_buf, "fetch --epoch "DF_U64" -v --single"
-				" --value %d\n", record->eph,
-				record->records[0].single.value);
+	if (rec_length != 0) {
+		if (record->records[0].snap == true)
+			sprintf(output_buf,
+				"fetch --tx %d -v --snap --recx \"%s\"\n",
+				record->eph, rec_buf);
 		else
-			sprintf(output_buf, "fetch --epoch "DF_U64" -v --recx"
-				" \"%s\"\n", record->eph, rec_buf);
+			sprintf(output_buf, "fetch --tx %d --recx \"%s\"\n",
+				record->eph, rec_buf);
 	}
+	return 0;
+}
+
+static int fetch_single(int index, int eph, struct extent *extents,
+	int extents_num, int rec_size, struct records *records,
+	char *output_buf)
+{
+	struct records *record;
+	char            rec_buf[512] = {0};
+	int             fetch_index;
+	int             rec_length = 0;
+	int             i;
+
+	D_ASSERT(index > 0);
 
+	fetch_index = rand() % index;
+	record      = &records[fetch_index];
+	for (i = 0; i < record->records_num; i++) {
+		struct array *array = &record->records[i].array;
+		int           length;
+		char          rec[64];
+
+		length = sprintf(rec, "[" DF_U64 ", " DF_U64 "]%d",
+				 array->extent.start, array->extent.end,
+				 array->value);
+
+		sprintf(rec_buf + rec_length, "%s ", rec);
+		rec_length += length + 1;
+	}
+
+	if (rec_length != 0) {
+		if (record->records[0].snap == true)
+			sprintf(output_buf,
+				"fetch --tx %d -v --snap --single --value %d\n",
+				record->eph, record->records[0].single.value);
+		else
+			sprintf(output_buf,
+				"fetch --tx %d --single --value %d\n",
+			    record->eph, record->records[0].single.value);
+
+	}
 	return 0;
 }
 
-int choose_op(int index)
+int choose_op(int index, int max_operation)
 {
 	if (index == 0)
 		return UPDATE_ARRAY;
 
 	/* FIXME: it should be able to specify the percentage
 	 * of each operation to generate the special workload.
 	 */
-	return rand() % MAX_OPS;
+	return rand() % max_operation;
 }
 
 struct operation {
-	int (*op)(int index, daos_epoch_t eph, struct extent *extents,
+	int (*op)(int index, int eph, struct extent *extents,
```

Comment:
int probably too shard for epoch for hlc, you may need uint64_t

Reply:
indeed